src/trade_executor.h

```diff
@@ -90,14 +105,43 @@ template <class T> class TradeExecutor {
         Trade<T>& trade = v_it->first;
         typename T::Ptr rsrc = v_it->second;
         if (rsrc->quantity() > cyclus::eps_rsrc()) {
+          // Get the original bid preference
+          double original_preference = trade.bid->preference();
+
+          // If the bid has NaN preference, use the request preference
+          if (std::isnan(original_preference)) {
+            original_preference = trade.request->preference();
+          }
+
+          // Start with the original preference as the adjusted preference
+          double adjusted_preference = original_preference;
+
+          // If we have access to the exchange context, use the adjusted
+          // preference that was actually used by the solver
+          if (ex_ctx) {
+            auto trader_it = ex_ctx->trader_prefs.find(trade.request->requester());
+            if (trader_it != ex_ctx->trader_prefs.end()) {
+              auto request_it = trader_it->second.find(trade.request);
+              if (request_it != trader_it->second.end()) {
+                auto bid_it = request_it->second.find(trade.bid);
+                if (bid_it != request_it->second.end()) {
+                  adjusted_preference = bid_it->second;
+                }
+              }
```

Comment:
(Not for this PR) Might want to make an issue to explore adding a method to the `ExchangeContext` to find something.  This may be the first time we've done this, but it seems like it might become useful in other cases.